### PR TITLE
update package.json engine

### PR DESCRIPTION
**Pull request by echo-bravo-yahoo**
more accurately describe the minimum functional version of node required to run the CLI.

also modifies the tests to run on our minimum supported node version to ensure we don't have regressions in v20.18 that don't show up in v20.x.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "typescript": "^5.6.3"
   },
   "engines": {
-    "node": ">=20.0.0"
+    "node": ">=20.18.0"
```
Comment (jrodewig) [inline package.json:59]: This conflicts with https://github.com/fauna/fauna-shell/pull/578. If we decide to go with this one, can we update the README as well?

Comment (echo-bravo-yahoo) [reply package.json:59]: yes, but i think #578 is a bit too prescriptivist. thoughts on:
```
## Requirements

- [Node.js](https://nodejs.org/en/download/package-manager) v20.18 or later.
  -  [Node.js](https://nodejs.org/en/download/package-manager) v22 or later recommended.
```

**Review by ecooper — APPROVED** [review]

**Review by jrodewig — APPROVED** [review]
